File: sdk/samples/all in on code/Visual Studio 2010/CSASPNETReverseAJAX/Dispatcher.asmx.cs
/****************************** Module Header ******************************\
* Module Name:    Dispatcher.asmx.cs
* Project:        CSASPNETReverseAJAX
* Copyright (c) Microsoft Corporation
*
* This web service is designed to be called by the Ajax client.
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
*
\*****************************************************************************/

using System.Web.Services;

namespace CSASPNETReverseAJAX
{
    /// <summary>
    /// This web service contains methods that help dispatching events to the client.
    /// </summary>
    [WebService(Namespace = "http://tempuri.org/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    [System.Web.Script.Services.ScriptService]
    public class Dispatcher : System.Web.Services.WebService
    {
        /// <summary>
        /// Dispatch the new message event.
        /// </summary>
        /// <param name="userName">The loged in user name</param>
        /// <returns>the message content</returns>
        [WebMethod]
        public string WaitMessage(string userName)
        {
            return ClientAdapter.Instance.GetMessage(userName);
        }
    }
}

File: sdk/samples/all in on code/Visual Studio 2010/VBASPNETReverseAJAX/Dispatcher.asmx.vb
'****************************** Module Header ******************************\
' Module Name:    Dispatcher.asmx.vb
' Project:        VBASPNETReverseAJAX
' Copyright (c) Microsoft Corporation
'
' This web service is designed to be called by the Ajax client.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
'
'*****************************************************************************/

Imports System.Web.Services

''' <summary>
''' This web service contains methods that help dispatching events to the client.
''' </summary>
<System.Web.Script.Services.ScriptService()> _
<System.Web.Services.WebService(Namespace:="http://tempuri.org/")> _
<System.Web.Services.WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<System.ComponentModel.ToolboxItem(False)> _
Public Class Dispatcher
    Inherits System.Web.Services.WebService

    ''' <summary>
    ''' Dispatch the new message event.
    ''' </summary>
    ''' <param name="userName">The loged in user name</param>
    ''' <returns>the message content</returns>
    <WebMethod()> _
    Public Function WaitMessage(ByVal userName As String) As String
        Return ClientAdapter.Instance.GetMessage(userName)
    End Function

End Class